tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/javainterop/InferredDependentlyTypeFunctionTest.java

```diff
@@ -158,9 +166,12 @@ public void testDependentlyTypedFunctionWithInferredTypedescValueNegative() {
                 145, 67);
         validateError(negativeResult, index++, "incompatible types: expected '(stream<boolean>|readonly)', found '" +
                 "(readonly|stream<int>|handle)'", 146, 34);
-        validateError(negativeResult, index++, "cannot infer type for parameter 'td'", 147, 25);
-        validateError(negativeResult, index++, "cannot infer type for parameter 'td'", 151, 13);
-        validateError(negativeResult, index++, "cannot infer type for parameter 'td'", 152, 13);
+        validateError(negativeResult, index++, "cannot infer the 'typedesc' argument for parameter 'td': " +
+                "expected a contextually-expected type or an argument for the parameter", 147, 25);
+        validateError(negativeResult, index++, "cannot infer the 'typedesc' argument for parameter 'td': " +
+                "expected a contextually-expected type or an argument for the parameter", 151, 13);
+        validateError(negativeResult, index++, "cannot infer the 'typedesc' argument for parameter 'td': " +
+                "expected a contextually-expected type or an argument for the parameter", 152, 13);
```

Comment:
There are tests with var, but there don't seem to be tests for non-var scenarios when there is no contextually-expected type? Like in the original issue?

```ballerina
fn();
```